pkg/identityserver/bunstore/contact_info_store.go

```diff
@@ -286,21 +286,45 @@ func (s *contactInfoStore) ValidateContactInfo(ctx context.Context, pb *ttnpb.Co
 	if len(pb.GetContactInfo()) != 1 {
 		return store.ErrValidationWithoutContactInfo.New()
 	}
-	contactInfo := pb.GetContactInfo()[0]
-	_, err := s.DB.NewUpdate().
-		Model(&ContactInfo{}).
-		Where("entity_type = ? AND entity_id = ?", pb.GetEntity().EntityType(), pb.GetEntity().IDString()).
-		Where("contact_method = ? AND LOWER(value) = LOWER(?)", contactInfo.GetContactMethod(), contactInfo.GetValue()).
+	contInfo := pb.GetContactInfo()[0]
+
+	entityType, entityUUID, err := s.getEntity(ctx, pb.GetEntity())
+	if err != nil {
+		return err
+	}
+
+	models, err := s.getContactInfoModelsBy(ctx, func(q *bun.SelectQuery) *bun.SelectQuery {
+		return q.
+			Where("entity_type = ? AND entity_id = ?", entityType, entityUUID).
+			Where("contact_method = ? AND LOWER(value) = LOWER(?)", contInfo.GetContactMethod(), contInfo.GetValue())
+	})
+	if err != nil {
+		return err
+	}
+	if len(models) != 1 {
```

Comment:
For the contact infos I don't think we do, speaking in a general term. 

But with the current way the method itself is used its guaranteed to be only one since its from a `ContactInfoValidation` which is fetched by providing `reference` and `token`

Reply:
Plus the login in the `ContactInfo` creation which stops having duplicate elements.

So there isn't an index that guarantees this but store logic pretty much guarantees it.